go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -59,12 +59,14 @@ func (bnc *BaseNetworkController) allocatePodIPsOnSwitch(pod *kapi.Pod,
 		if util.PodTerminating(pod) {
 			klog.Infof("Ignoring IP allocation for terminating pod: %s/%s, on deleted "+
 				"node: %s", pod.Namespace, pod.Name, pod.Spec.NodeName)
-			return expectedLogicalPortName, nil
 		} else if bnc.doesNetworkRequireIPAM() {
 			// unknown condition how we are getting a non-terminating pod without a node here
 			klog.Errorf("Pod IP allocation found for a non-existent node in API with unknown "+
 				"condition. Pod: %s/%s, node: %s", pod.Namespace, pod.Name, pod.Spec.NodeName)
+		} else {
+			klog.Infof("Pod %s/%s is running on a node %s that does not exist", pod.Namespace, pod.Name, pod.Spec.NodeName)
 		}
+		return expectedLogicalPortName, nil
```

Comment:
return expectedLogicalPortName, fmt.Errorf("pod: %s/%s is in a non-existing node: %s: %w", err)
[5:35](https://redhat-internal.slack.com/archives/D027TCD5A75/p1696368952096229)
do not return expectedLogicalPortName, nil because that means it did not fail, right?